packages/lime-proto-babeld/files/usr/lib/lua/lime/proto/babeld.lua

```diff
@@ -99,25 +111,74 @@ function babeld.setup_interface(ifname, args)
 
 	utils.log("lime.proto.babeld.setup_interface(...)", ifname)
 
-	local vlanId = args[2] or 17
+	local vlanId = tonumber(args[2]) or 17
 	local vlanProto = args[3] or "8021ad"
 	local nameSuffix = args[4] or "_babeld"
 
+
+	--! If Babeld is without VLAN (vlanId is 0) it should run directly on plain
+	--! ethernet interfaces, but the ones which are inside of the LAN bridge
+	--! (e.g. eth0 or eth0.1) cannot have an IPv6 Link-Local and Babeld needs it.
+	--! So Babeld has to run on the bridge interface br-lan
+	local isIntoLAN = false
+	local addIPtoIf = true
+	for _,v in pairs(args["deviceProtos"]) do
+		if v == "lan" then
+			isIntoLAN = true
+		--! would be weird to add a static IP to the WAN interface
+		elseif v == "wan" then
+			addIPtoIf = false
+		end
+	end
+
+	if ifname:match("^wlan") then
+		--! currently (2019-10-12) mode-ap and mode-apname have an hardcoded
+		--! "option network lan" so they are always in the br-lan bridge
+		if ifname:match("^wlan.*ap$") or ifname:match("^wlan.*apname$") then
```

Comment:
At the first part of this function there is the following code:
```
	if not args["specific"] and ifname:match("^wlan%d+.ap") then
		utils.log("lime.proto.babeld.setup_interface(...)", ifname, "ignored")
		return
	end
```
As far as I understand for ap and apname it won't reach this part of the code. Is it ok?

Reply:
Yes, by default the AP and APname interfaces are not used for the backbone and the routing.
For using an AP interface for routing with Babeld, APbb should be used, as implemented in #554 and documented in lime-example [here](https://github.com/libremesh/lime-packages/blob/18c8aa59ec14bf8a2e82f72b0b761194d525213c/packages/lime-docs/files/lime-example#L134-L168).

Reply:
Ok maybe I am not understanding. In which conditions the program will get into the if of line 137?

Reply:
You mean this if, right?

https://github.com/libremesh/lime-packages/blob/a8369d928e34d00bad8d245c2a486688f3c78b11/packages/lime-proto-babeld/files/usr/lib/lua/lime/proto/babeld.lua#L137
It gets hit when an AP or an APname is configured using interface-specific configuration (some documentation [on the website here](https://libremesh.org/docs/en_change_wifi_behavior.html)) and babeld is also selected in the interface-specific configuration.

An example of a case where configuring an AP with interface-specific configuration is [here](https://github.com/libremesh/lime-packages/issues/262#issuecomment-538527314) in #262 (5 GHz radio used just for mesh and 2.4 GHz radio used for AP+APname+mesh, which cannot currently be done with general configuration).

Something like: (the non-specified options in the wifi sections are taken from the generic wifi configuration, see db1c3505ac39b178ed19cb0ecf939ce30e000032. I don't know if more options should be specified for the net sections?)

```
config wifi radio0 # assuming that radio0 is 2.4 GHz
	list modes 'ap'
	list modes 'apname'
	list modes 'ieee80211s'

config net wireless2ap
	option linux_name 'wlan0-ap'
	list protocols lan # this is hardcoded anyway in ap proto, adding also here just for clarity
	list protocols babeld:17

config net wireless2apname
	option linux_name 'wlan0-apname'
	list protocols lan # this is hardcoded anyway in apname proto, adding also here just for clarity
	list protocols babeld:17

config net wireless2mesh
	option linux_name 'wlan0-mesh'
	list protocols babeld:17

config wifi radio1 # assuming that radio1 is 5 GHz
	list modes 'ieee80211s'

config net wireless5mesh
	option linux_name 'wlan1-mesh'
	list protocols babeld:17
```

(beware, I did not test this example)